test/Renci.SshNet.Tests/Common/AsyncSocketListener.cs

```diff
@@ -322,23 +284,36 @@ void ConnectionDisconnected()
                 }
                 catch (ObjectDisposedException)
                 {
-                    // TODO On .NET 7, sometimes we get ObjectDisposedException when _started but only on appveyor, locally it works
-                    ConnectionDisconnected();
```

Comment:
This was the main problem. If we have received ObjectDisposedException, we don't want to call `Shutdown` on the socket because we will get another ObjectDisposedException, and that will crash the process.